lib/Targets/AIETargetXAIEV2.cpp

```diff
@@ -330,12 +330,15 @@ mlir::LogicalResult AIETranslateToXAIEV2(ModuleOp module, raw_ostream &output) {
         if (block.getNumSuccessors() > 0) {
           Block *nextBlock = block.getSuccessors()[0]; // should have only one
                                                        // successor block
+          int enableNextBd = 1;
+          if (!nextBlock->getOps<EndOp>().empty())
+            enableNextBd = 0;
+
           int nextBdNum = blockMap[nextBlock];
           output << "XAie_DmaSetNextBd(" << tileDMAInstRefStr(col, row, bdNum)
                  << ", "
                  << " /* nextbd */ " << nextBdNum << ", "
-                 << " /* enableNextBd */ 1);\n"; // TODO Check if br ^end: to
-                                                 // disable this?
+                 << " /* enableNextBd */ " << enableNextBd << ");\n";
```

Comment:
It would be good to include a simple test to check for 0 and 1 for just the mem blocks with cr.br ^end and cf.br ^bd0 respectively